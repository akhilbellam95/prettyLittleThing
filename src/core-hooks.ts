import { configureStore, createStore } from "@reduxjs/toolkit";
import { useSelector } from "react-redux";
import { TypedUseSelectorHook, useDispatch } from "react-redux";
import { CartState } from "./Features/Cart/cartSlice";
import { ProductsState } from "./Features/Product/productSlice";

export interface ApplicationState {
  cart: CartState;
  product: ProductsState;
}

export const useAppDispatch = () =>
  useDispatch<ReturnType<typeof configureStore>['dispatch']>();

export const useAppSelector: TypedUseSelectorHook<ApplicationState> =
  useSelector;