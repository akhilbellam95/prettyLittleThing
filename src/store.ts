// eslint-disable-next-line prettier/prettier
import { configureStore } from '@reduxjs/toolkit';
import cartReducer from './Features/Cart/cartSlice';
import productReducer from './Features/Product/productSlice';

export const store = configureStore({
  reducer: {
    cart: cartReducer,
    product: productReducer,
  },
  devTools: process.env.NODE_ENV !== 'production',
});

export type AppDispatch = typeof store.dispatch;

export type RootState = ReturnType<typeof store.getState>;
