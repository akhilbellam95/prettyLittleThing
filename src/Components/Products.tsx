import { FlatList, StyleSheet, View } from 'react-native';
import React, { useEffect } from 'react';
import { getProducts } from '../Features/Product/productSlice';
import { ApplicationState, useAppDispatch, useAppSelector } from '../core-hooks';
import { ProductItem } from './ProductItem';

export interface Product {
  id: number;
  name: string;
  img: string;
  price: number;
  colour: string;
}

interface Products {
  products: Product[];
}

const Products = () => {
  const { products } = useAppSelector((state: ApplicationState) => state.product);
  const dispatch = useAppDispatch();

  useEffect(() => {
    dispatch(getProducts());
  }, []);

  return (
    <View>
      <FlatList data={products} renderItem={({ item }) => <ProductItem product={item} />} />
    </View>
  );
};

export default Products;

export const styles = StyleSheet.create({
  alignCenter: {
    alignItems: 'center',
  },
  button: {
    backgroundColor: 'yellow',
    width: 120,
    height: 40,
    justifyContent: 'center',
    position: 'relative',
    paddingHorizontal: 20,
    borderWidth: 1,
    borderColor: 'black',
  },
  card: {
    padding: 20,
    borderWidth: 1,
    marginHorizontal: 20,
    marginVertical: 10,
  },
  productDetails: {
    borderTopWidth: 1,
    borderTopColor: 'black',
    marginTop: 20,
    paddingTop: 15,
  },
  productImage: {
    width: 280,
    height: 500,
  },
  label: {
    width: 60,
    marginBottom: 5,
  },
  row: {
    flexDirection: 'row',
  },
  seperator: {
    height: 20,
  },
  textCenter: {
    textAlign: 'center',
  },
  value: {
    width: '75%',
  },
});
