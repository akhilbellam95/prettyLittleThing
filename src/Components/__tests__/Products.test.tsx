import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { Provider } from 'react-redux';
import Products from '../Products';
import { useAppDispatch, useAppSelector } from '../../core-hooks';
import { mockCartItems, mockProducts } from '../../mocks/data';
import { act } from 'react-test-renderer';
import { store } from '../../store';

jest.mock('../../core-hooks', () => ({
  useAppDispatch: jest.fn(),
  useAppSelector: jest.fn(),
}));

describe('Products component testing', () => {
  const mockDispatch = jest.fn();
  const mockSelector = jest.fn();

  beforeAll(() => {
    (useAppSelector as jest.Mock).mockImplementation(mockSelector);
    (useAppDispatch as jest.Mock).mockReturnValue(mockDispatch);
  });

  beforeEach(() => {
    mockSelector.mockReturnValue({
      products: mockProducts,
      cartItems: mockCartItems,
    });
  });

  it('renders correctly', () => {
    const snap: unknown = render(
      <Provider store={store}>
        <Products />
      </Provider>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
