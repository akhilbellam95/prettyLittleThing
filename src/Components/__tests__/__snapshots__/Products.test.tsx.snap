// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products component testing renders correctly 1`] = `
<View>
  <RCTScrollView
    data={
      [
        {
          "colour": "Black",
          "id": 1,
          "img": "http://cdn-img.prettylittlething.com/9/0/a/a/90aa90903a135ee59594f47c7685aa7ef3046e44_cly8063_1.jpg?imwidth=1024",
          "name": "Black Sheet Strappy Textured Glitter Bodycon Dress",
          "price": 10,
        },
        {
          "colour": "Stone",
          "id": 2,
          "img": "https://cdn-img.prettylittlething.com/3/6/5/a/365a5d1dce6a2b77b564379b302c9d83afccf33b_cmd2051_1.jpg?imwidth=1024",
          "name": "Stone Ribbed Strappy Cut Out Detail Bodycon Dress",
          "price": 4,
        },
        {
          "colour": "Black",
          "id": 3,
          "img": "https://cdn-img.prettylittlething.com/d/c/3/3/dc337260f9ecefdb99a8c8e98cd73ccb1b79cea5_cmb6804_4.jpg?imwidth=1024",
          "name": "Black Frill Tie Shoulder Bodycon Dress",
          "price": 7.99,
        },
        {
          "colour": "Red",
          "id": 5,
          "img": "https://cdn-img.prettylittlething.com/f/7/1/8/f718a4011ddf92f48aeefff6da0f475178694599_cly0842_1.jpg?imwidth=1024",
          "name": "Red Pin Stripe Belt T Shirt Dress",
          "price": 17,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderWidth": 1,
              "marginHorizontal": 20,
              "marginVertical": 10,
              "padding": 20,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
              }
            }
          >
            <View
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "height": 500,
                    "width": 280,
                  },
                ]
              }
            >
              <FastImageView
                defaultSource={null}
                resizeMode="contain"
                source={
                  {
                    "priority": "normal",
                    "uri": "http://cdn-img.prettylittlething.com/9/0/a/a/90aa90903a135ee59594f47c7685aa7ef3046e44_cly8063_1.jpg?imwidth=1024",
                  }
                }
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              {
                "borderTopColor": "black",
                "borderTopWidth": 1,
                "marginTop": 20,
                "paddingTop": 15,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "marginBottom": 5,
                    "width": 60,
                  }
                }
              >
                Name: 
              </Text>
              <Text
                style={
                  {
                    "width": "75%",
                  }
                }
              >
                Black Sheet Strappy Textured Glitter Bodycon Dress
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "marginBottom": 5,
                    "width": 60,
                  }
                }
              >
                Price: 
              </Text>
              <Text
                style={
                  {
                    "width": "75%",
                  }
                }
              >
                10
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "marginBottom": 5,
                    "width": 60,
                  }
                }
              >
                Colour: 
              </Text>
              <Text
                style={
                  {
                    "width": "75%",
                  }
                }
              >
                Black
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "height": 20,
              }
            }
          />
          <View
            style={
              {
                "alignItems": "flex-end",
              }
            }
          >
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "borderColor": "black",
                    "borderWidth": 1,
                    "width": 120,
                  },
                ]
              }
            >
              <View
                style={
                  {
                    "height": 20,
                  }
                }
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "backgroundColor": "yellow",
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
                testID="remove-btn"
              >
                <Text
                  style={
                    {
                      "textAlign": "center",
                    }
                  }
                >
                  -
                </Text>
              </View>
              <View
                style={
                  {
                    "height": 40,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    {
                      "textAlign": "center",
                      "width": 40,
                    }
                  }
                >
                  2
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "backgroundColor": "yellow",
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
                testID="add-btn"
              >
                <Text
                  style={
                    {
                      "textAlign": "center",
                    }
                  }
                >
                  +
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderWidth": 1,
              "marginHorizontal": 20,
              "marginVertical": 10,
              "padding": 20,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
              }
            }
          >
            <View
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "height": 500,
                    "width": 280,
                  },
                ]
              }
            >
              <FastImageView
                defaultSource={null}
                resizeMode="contain"
                source={
                  {
                    "priority": "normal",
                    "uri": "https://cdn-img.prettylittlething.com/3/6/5/a/365a5d1dce6a2b77b564379b302c9d83afccf33b_cmd2051_1.jpg?imwidth=1024",
                  }
                }
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              {
                "borderTopColor": "black",
                "borderTopWidth": 1,
                "marginTop": 20,
                "paddingTop": 15,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "marginBottom": 5,
                    "width": 60,
                  }
                }
              >
                Name: 
              </Text>
              <Text
                style={
                  {
                    "width": "75%",
                  }
                }
              >
                Stone Ribbed Strappy Cut Out Detail Bodycon Dress
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "marginBottom": 5,
                    "width": 60,
                  }
                }
              >
                Price: 
              </Text>
              <Text
                style={
                  {
                    "width": "75%",
                  }
                }
              >
                4
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "marginBottom": 5,
                    "width": 60,
                  }
                }
              >
                Colour: 
              </Text>
              <Text
                style={
                  {
                    "width": "75%",
                  }
                }
              >
                Stone
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "height": 20,
              }
            }
          />
          <View
            style={
              {
                "alignItems": "flex-end",
              }
            }
          >
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "borderColor": "black",
                    "borderWidth": 1,
                    "width": 120,
                  },
                ]
              }
            >
              <View
                style={
                  {
                    "height": 20,
                  }
                }
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "backgroundColor": "yellow",
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
                testID="remove-btn"
              >
                <Text
                  style={
                    {
                      "textAlign": "center",
                    }
                  }
                >
                  -
                </Text>
              </View>
              <View
                style={
                  {
                    "height": 40,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    {
                      "textAlign": "center",
                      "width": 40,
                    }
                  }
                >
                  1
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "backgroundColor": "yellow",
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
                testID="add-btn"
              >
                <Text
                  style={
                    {
                      "textAlign": "center",
                    }
                  }
                >
                  +
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderWidth": 1,
              "marginHorizontal": 20,
              "marginVertical": 10,
              "padding": 20,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
              }
            }
          >
            <View
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "height": 500,
                    "width": 280,
                  },
                ]
              }
            >
              <FastImageView
                defaultSource={null}
                resizeMode="contain"
                source={
                  {
                    "priority": "normal",
                    "uri": "https://cdn-img.prettylittlething.com/d/c/3/3/dc337260f9ecefdb99a8c8e98cd73ccb1b79cea5_cmb6804_4.jpg?imwidth=1024",
                  }
                }
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              {
                "borderTopColor": "black",
                "borderTopWidth": 1,
                "marginTop": 20,
                "paddingTop": 15,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "marginBottom": 5,
                    "width": 60,
                  }
                }
              >
                Name: 
              </Text>
              <Text
                style={
                  {
                    "width": "75%",
                  }
                }
              >
                Black Frill Tie Shoulder Bodycon Dress
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "marginBottom": 5,
                    "width": 60,
                  }
                }
              >
                Price: 
              </Text>
              <Text
                style={
                  {
                    "width": "75%",
                  }
                }
              >
                7.99
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "marginBottom": 5,
                    "width": 60,
                  }
                }
              >
                Colour: 
              </Text>
              <Text
                style={
                  {
                    "width": "75%",
                  }
                }
              >
                Black
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "height": 20,
              }
            }
          />
          <View
            style={
              {
                "alignItems": "flex-end",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "height": 40,
                  "justifyContent": "center",
                  "paddingHorizontal": 20,
                  "position": "relative",
                  "width": 120,
                }
              }
              testID="add-to-cart"
            >
              <Text
                style={
                  {
                    "textAlign": "center",
                  }
                }
              >
                Add to Cart
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderWidth": 1,
              "marginHorizontal": 20,
              "marginVertical": 10,
              "padding": 20,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
              }
            }
          >
            <View
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "height": 500,
                    "width": 280,
                  },
                ]
              }
            >
              <FastImageView
                defaultSource={null}
                resizeMode="contain"
                source={
                  {
                    "priority": "normal",
                    "uri": "https://cdn-img.prettylittlething.com/f/7/1/8/f718a4011ddf92f48aeefff6da0f475178694599_cly0842_1.jpg?imwidth=1024",
                  }
                }
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              {
                "borderTopColor": "black",
                "borderTopWidth": 1,
                "marginTop": 20,
                "paddingTop": 15,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "marginBottom": 5,
                    "width": 60,
                  }
                }
              >
                Name: 
              </Text>
              <Text
                style={
                  {
                    "width": "75%",
                  }
                }
              >
                Red Pin Stripe Belt T Shirt Dress
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "marginBottom": 5,
                    "width": 60,
                  }
                }
              >
                Price: 
              </Text>
              <Text
                style={
                  {
                    "width": "75%",
                  }
                }
              >
                17
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "marginBottom": 5,
                    "width": 60,
                  }
                }
              >
                Colour: 
              </Text>
              <Text
                style={
                  {
                    "width": "75%",
                  }
                }
              >
                Red
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "height": 20,
              }
            }
          />
          <View
            style={
              {
                "alignItems": "flex-end",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "height": 40,
                  "justifyContent": "center",
                  "paddingHorizontal": 20,
                  "position": "relative",
                  "width": 120,
                }
              }
              testID="add-to-cart"
            >
              <Text
                style={
                  {
                    "textAlign": "center",
                  }
                }
              >
                Add to Cart
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
